src/mplhep/plot.py

```diff
@@ -167,11 +172,85 @@ def histplot(
         else get_histogram_axes_title(hists[0].axes[0])
     )
 
-    # Cast to plottables
-    plottables = [
-        Plottable(h.values(), edges=final_bins, variances=h.variances()) for h in hists
-    ]
+    # Show under/overflow bins
+    ## check underflow/overflow bin exist
+
+    underflow, overflow = 0.0, 0.0
 
+    for h in hists:
+        if (
+            not isinstance(h, Hist.hist.Hist)
+            and not "uproot.models.TH" in str(type(h))
+            and flow is not None
+        ):
+            continue
+        elif isinstance(h, Hist.hist.Hist) and len(h.values(flow=True)) < len(
+            final_bins
+        ):
+            continue
+        else:
+            underflow = underflow + h.values(flow=True)[0]
+            overflow = overflow + h.values(flow=True)[-1]
+
+    # "show": Add additional bin with 5 times bin width
+    plottables = []
+    flow_bins = final_bins
+    for i, h in enumerate(hists):
+        value, variance = h.values(), h.variances()
+        if not isinstance(h, Hist.hist.Hist) and not "uproot.models.TH" in str(type(h)):
+            print(f"Warning: {type(h)} is not allowed to get flow bins")
```

Comment:
Hmm, is this triggered every time you pass just an array as a hist?

Reply:
Seems you have this solved for 2D with `and flow is not None`

Reply:
This should be triggered every time, otherwise the hist would not filled as `flow` set to `None` after 1 iteration

Reply:
Is this really a good practice tho? I don't think I should see this warning if I all I do is call `histplot([1,2,3,2,1])` which I think will happen here right?

Reply:
Right, add `and flow is not None` and indeed if we have `None` option, we would keep the case `plottables.append(Plottable(value, edges=final_bins, variances=variance))`